http_parser.c

```diff
@@ -472,6 +483,92 @@ static struct {
 
 int http_message_needs_eof(const http_parser *parser);
 
+const char* find_crlf(const char* p, const char* data, size_t len);
+
+
+#if defined(__SSE2__) && defined(__GNUC__) 
+
+#include <emmintrin.h>
+
+const char* find_crlf(const char* p, const char* data, size_t len) {
+  const char* lastp = MIN(data + len, HTTP_MAX_HEADER_SIZE + p);
+
+  int32_t result = 0;
+  __m128i v1, v2;
+  /* [ c, 0, 0, 0, 0, 0 .. 0 ] */
+  __m128i vCR = _mm_set_epi32(0x0a0a0a0a, 0x0a0a0a0a,0x0a0a0a0a, 0x0a0a0a0a);
```

Comment:
`_mm_set1_epi32(0x0a0a0a0a)` does the same thing and compiles to one fewer instruction:

```asm
  vmovdqa xmm0, XMMWORD PTR .LC0[rip]
  vmovaps XMMWORD PTR vCR[rip], xmm0
```
vs just:
```asm
  vmovaps XMMWORD PTR vCR[rip], xmm0
```